src/Analyzers/CSharp/Analyzers/UseAutoProperty/CSharpUseAutoPropertyAnalyzer.cs

```diff
@@ -213,4 +252,29 @@ private static bool CheckExpressionSyntactically(ExpressionSyntax expression)
             ? fieldDeclaration
             : variableDeclarator;
     }
+
+    protected override void AddAccessedFields(
+        SemanticModel semanticModel,
+        IMethodSymbol accessor,
+        HashSet<string> fieldNames,
+        HashSet<IFieldSymbol> result,
+        CancellationToken cancellationToken)
```

Comment:
new helper that analyzed the code inside an accessor to see which fields (from teh containing type) that were accessed.  These fields will then be checked to see if any could be replaced with `field`